src/SerialLibrary/__init__.py

```diff
@@ -524,12 +525,50 @@ def read_until(self, terminator=LF, size=None, encoding=None, port_locator=None)
         """
         if size is not None:
             size = float(size)
-        if terminator != LF:
+        if terminator != LF and not isinstance(terminator, (bytes, bytearray)):
             terminator = self._encode(terminator)
         return self._decode(
             self._port(port_locator).read_until(terminator=terminator, size=size),
             encoding)
 
+    def read_until_pattern(self, pattern_regexp, terminator=LF, size=None, encoding=None, port_locator=None):
```

Comment:
I'd rename `pattern_regexp` to `pattern` to avoid confusion. It should be of type `str`.